### ISSUE TITLE
Variance is wrong from `rinvgamma`

### ISSUE THREAD

**Issue by dklinenberg2020** [dkahle/invgamma#2]
I am interested in simulation $x \sim \Gamma^{-1}(shape=7,rate=2.01)$, following `invgamma` notation.

Theoretically, I should expect to get $\mathbb{E}[x]=\frac{7}{2.01-1}\approx7$ and $Var(x)=\frac{7^2}{(2.01-1)^2(2.01-2)}\approx 4800$.

I  run the following line: `ig<-invgamma::rinvgamma(n=100000,shape = shape,rate = scale)`

Then `mean(ig)`=6.92 and `var(ig)`=208, which is much too small for the variance. 

Any ideas about what's going on?

**Comment by dkahle** [comment]
Hey @dklinenberg2020 . I think this is a parameterization issue. Updating docs to clarify. 

Does the below look right to you? Remember that the rate here is referring to the rate of the respective gamma distribution.

``` r
library("invgamma")

shape <- 7; rate <- 2.01
x <- rinvgamma(1e6, shape, rate)

mean(x) 
#> [1] 0.3353017
rate / (shape - 1)
#> [1] 0.335

var(x) 
#> [1] 0.02263876
rate^2 / ( (shape - 1)^2 * (shape - 2) )
#> [1] 0.022445
```

<sup>Created on 2023-01-13 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

![image](https://user-images.githubusercontent.com/424139/212407301-2df95d77-b7ca-41f2-a1be-ab850a700563.png)

**Comment by dklinenberg2020** [comment]
Hi David,

This is quite embarrassing. In my question above, I flipped the rate and shape parameters in my question. You are most certainly correct, I had set the rate and shape backward in my question.

It was supposed to say `invgamma::rinvgamma(n=1e6,rate = 7,shape = 2.01)`.

Running this, instead, will lead to $\mathbb{E}[x]\approx 7$ and $Var(x) \approx 280$. The mean is what we'd expect, but the variance is much smaller. Theoretically, we'd expect $Var(x)\approx4800$.

Sorry for the mess up in the question! Your answer perfectly answered my incorrectly worded question above.
